Mobile Buy SDK Sample Apps/Advanced App - ObjC/Mobile Buy SDK watchOS Sample App Extension/ProductItem.swift

```diff
@@ -25,18 +25,54 @@
 //
 
 import Foundation
+import Buy
+import WatchKit
 
 class ProductItem {
     
     var index: Int = 0
+    var product: BUYProduct!
     var productsModel: ProductsModel!
+    var variants: [BUYProductVariant?] = []
     
-    init(index: Int, productsModel: ProductsModel) {
+    init(index: Int, product: BUYProduct, productsModel: ProductsModel) {
         self.index = index
         self.productsModel = productsModel
+        self.product = product
+    }
+    
+    func variantPrice(index: Int) -> String {
+        return self.price(variant: self.variants[index])
     }
     
     func configure(controller: ProductDetailsInterfaceController) {
         self.productsModel.configure(interfaceController: controller, index: self.index)
+
+        if self.product.optionsArray().count > 1 {
+            controller.applePayButton.setHidden(true)
+            controller.variantPicker.setHidden(true)
+            controller.topSeparator.setHidden(true)
+        } else {
+            self.variants = self.product.variantsArray()
+            let caption = String(format: "%@:", (self.product.optionsArray().first?.name)!)
```

Comment:
Well in this case it doesn't matter since we're only going to show variants if there's only one option. If there's more than one option, the user will have to see the product on their phone.